source/snippets/registry-index.txt

```diff
@@ -0,0 +1,74 @@
+
+.. _snip-manual-reg-index:
+
+=====================================
+Create a Registry Index File Manually 
+=====================================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. warning:: Experimental feature
+
+   .. include:: /includes/warnings/warn-experimental-feature.rst
+
+This page discusses how to manually create a registry index file. To
+generate the registry index file using a script, see :ref:`Create a
+Registry Index File <snip-make-a-registry-index>`. 
+
+To create the registry index file:
+
+#. Copy the following TypeScript template and save it as ``index.bson``:
```

Comment:
Uh, so … this is the typescript *definitions* for what `index.bson` contains, but it’s not the actual contents. `index.bson` is actual BSON, i.e. binary data, not text.

I think that as this is for fairly advanced users only, it’s okay to leave out things like how to come up with the actual contents of the file (e.g. in the case of the default repository, the information is grabbed from the individual snippets’ package.json files).

I’d suggest adding 

```js
import bson from 'bson';
import zlib from 'zlib';
``` 

lines at the top of the typescript example here, and then adding the following at the bottom:

```js
const indexFileContents: SnippetIndexFile = {
   indexFileVersion: 1,
   index: [ /* ... */ ],
   metadata: { homepage: 'https://example.com' }
};

// Serialize, compress and store the index file:
fs.writeFileSync('index.bson.br',
   zlib.brotliCompressSync(
      bson.serialize(indexFileContents)));
```

The file can then be store as e.g. `make-index.ts`, and can be run as `ts-node make-index.ts` in order to actually create the `index.bson.br` file.

Reply:
Done